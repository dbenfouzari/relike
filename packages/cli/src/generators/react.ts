import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
