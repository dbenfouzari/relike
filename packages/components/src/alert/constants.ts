import Color from "../color";
import Icons, { IconData } from "../icons";

export enum AlertSeverities {
  error = "error",
  warning = "warning",
  info = "info",
  success = "success",
}

export const DEFAULT_ICONS: Record<AlertSeverities, IconData> = {
  [AlertSeverities.error]: Icons.error_outline,
  [AlertSeverities.warning]: Icons.report_problem_outlined,
  [AlertSeverities.info]: Icons.info_outline,
  [AlertSeverities.success]: Icons.check_circle_outline,
};

export const DEFAULT_COLORS: Record<AlertSeverities, Record<"font" | "icon" | "background", Color>> = {
  [AlertSeverities.error]: {
    font: new Color(0xff5f2120),
    icon: new Color(0xffef5350),
    background: new Color(0xfffdeded),
  },
  [AlertSeverities.warning]: {
    font: new Color(0xff663c00),
    icon: new Color(0xffff9900),
    background: new Color(0xfffff4e5),
  },
  [AlertSeverities.info]: {
    font: new Color(0xff014361),
    icon: new Color(0xff03a9f4),
    background: new Color(0xffe5f6fd),
  },
  [AlertSeverities.success]: {
    font: new Color(0xff1e4620),
    icon: new Color(0xff4caf4f),
    background: new Color(0xffedf7ed),
  },
};

export type AvatarIcon = keyof typeof DEFAULT_ICONS;
export type AvatarColors = typeof DEFAULT_COLORS[keyof typeof DEFAULT_COLORS];
