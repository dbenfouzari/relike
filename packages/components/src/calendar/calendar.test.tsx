import { act, fireEvent, render } from "@testing-library/react";

import DateTime from "../date-time";
import Duration from "../duration";
import Calendar from "./calendar";
import { MONTHS } from "./constants";

const currDateTime = new DateTime({ year: 2021, month: DateTime.august, day: 26, hour: 21, minute: 14 });

describe("Calendar", () => {
  it("should render successfully", () => {
    const { container } = render(<Calendar initialValue={currDateTime} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should handle previous month", async () => {
    const { container, getByTestId } = render(
      <Calendar
        initialValue={currDateTime}
        events={[
          { date: currDateTime.subtract(Duration.days(60)), title: "Past Event" },
          { date: currDateTime, title: "Storybook Event That Rocks !" },
        ]}
      />,
    );

    const currentMonth = currDateTime.month;
    const currentMonthIndex = currentMonth - 1;

    const month = getByTestId("month");
    const currMonth = MONTHS.en[currentMonthIndex];
    const prevCurrMonth = MONTHS.en[currentMonthIndex - 1];
    const nextCurrMonth = MONTHS.en[currentMonthIndex + 1];

    expect(month.textContent).toEqual(currMonth);
    expect(container.firstChild).toMatchSnapshot();

    const prevMonth = getByTestId("prev-month");
    const nextMonth = getByTestId("next-month");

    act(() => {
      fireEvent.click(prevMonth);
    });

    expect(month.textContent).toEqual(prevCurrMonth);
    expect(container.firstChild).toMatchSnapshot();

    act(() => {
      fireEvent.click(nextMonth);
      fireEvent.click(nextMonth);
    });

    expect(month.textContent).toEqual(nextCurrMonth);
    expect(container.firstChild).toMatchSnapshot();
  });
});
