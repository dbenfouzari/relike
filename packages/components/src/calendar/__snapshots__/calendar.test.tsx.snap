// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar should handle previous month 1`] = `
<div
  class="calendar"
>
  <div
    class="top"
  >
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="prev-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_left
      </span>
    </button>
    <div
      class="center"
    >
      <span
        data-testid="month"
      >
        august
      </span>
      <span>
        2021
      </span>
    </div>
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="next-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_right
      </span>
    </button>
  </div>
  <div
    class="inner"
  >
    <span
      class="day_name"
    >
      <span>
        mon
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        tue
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        wed
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        thu
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        fri
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sat
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sun
      </span>
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        26
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        27
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        31
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        4
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        5
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        6
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        7
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        8
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        9
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        10
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        11
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        12
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        13
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        14
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        15
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        16
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        17
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        18
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        19
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        20
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        21
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        22
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        23
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        24
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        25
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        26
      </span>
      <div
        class="events"
      >
        <div
          class="wrapper"
        >
          <span>
            Storybook Event That Rocks !
          </span>
          <span
            class="hour"
          >
            21
            :
            14
          </span>
        </div>
      </div>
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        27
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number day_number__current"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        31
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        4
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        5
      </span>
      <div
        class="events"
      />
    </span>
  </div>
</div>
`;

exports[`Calendar should handle previous month 2`] = `
<div
  class="calendar"
>
  <div
    class="top"
  >
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="prev-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_left
      </span>
    </button>
    <div
      class="center"
    >
      <span
        data-testid="month"
      >
        july
      </span>
      <span>
        2021
      </span>
    </div>
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="next-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_right
      </span>
    </button>
  </div>
  <div
    class="inner"
  >
    <span
      class="day_name"
    >
      <span>
        mon
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        tue
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        wed
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        thu
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        fri
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sat
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sun
      </span>
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        4
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        5
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        6
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        7
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        8
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        9
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        10
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        11
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        12
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        13
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        14
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        15
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        16
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        17
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        18
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        19
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        20
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        21
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        22
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        23
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        24
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        25
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        26
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        27
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        31
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
  </div>
</div>
`;

exports[`Calendar should handle previous month 3`] = `
<div
  class="calendar"
>
  <div
    class="top"
  >
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="prev-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_left
      </span>
    </button>
    <div
      class="center"
    >
      <span
        data-testid="month"
      >
        september
      </span>
      <span>
        2021
      </span>
    </div>
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="next-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_right
      </span>
    </button>
  </div>
  <div
    class="inner"
  >
    <span
      class="day_name"
    >
      <span>
        mon
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        tue
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        wed
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        thu
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        fri
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sat
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sun
      </span>
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        31
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        4
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        5
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        6
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        7
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        8
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        9
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        10
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        11
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        12
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        13
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        14
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        15
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        16
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        17
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        18
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        19
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        20
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        21
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        22
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        23
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        24
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        25
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        26
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        27
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
  </div>
</div>
`;

exports[`Calendar should render successfully 1`] = `
<div
  class="calendar"
>
  <div
    class="top"
  >
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="prev-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_left
      </span>
    </button>
    <div
      class="center"
    >
      <span
        data-testid="month"
      >
        august
      </span>
      <span>
        2021
      </span>
    </div>
    <button
      class="sc-gtsrHT dtPxGD"
      data-testid="next-month"
    >
      <span
        class="material-icons sc-bdnxRM cbDTns"
      >
        arrow_right
      </span>
    </button>
  </div>
  <div
    class="inner"
  >
    <span
      class="day_name"
    >
      <span>
        mon
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        tue
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        wed
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        thu
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        fri
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sat
      </span>
    </span>
    <span
      class="day_name"
    >
      <span>
        sun
      </span>
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        26
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        27
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        31
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        4
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        5
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        6
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        7
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        8
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        9
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        10
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        11
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        12
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        13
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        14
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        15
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        16
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        17
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        18
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        19
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        20
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        21
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        22
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        23
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        24
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        25
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        26
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        27
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number day_number__current"
      >
        28
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        29
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        30
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell"
    >
      <span
        class="day_number"
      >
        31
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        1
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        2
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        3
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        4
      </span>
      <div
        class="events"
      />
    </span>
    <span
      class="cell cell__out_of_range"
    >
      <span
        class="day_number"
      >
        5
      </span>
      <div
        class="events"
      />
    </span>
  </div>
</div>
`;
